src/communication/communicator.hpp

```diff
@@ -57,10 +57,10 @@ class communicator {
         struct gid_info {
             using connection_list = decltype(std::declval<recipe>().connections_on(0));
             cell_gid_type gid;       // global identifier of cell
-            cell_gid_type domain_id; // index of cell in this domain
+            cell_gid_type index_on_domain; // index of cell in this domain
```

Comment:
Would still prefer this to be a `cell_size_type` — if we can keep `cell_gid_type` for cell global ids, then we have the benefit of type-driven documentation, which I really, really like.

Reply:
Done.